### DISCUSSION TITLE
How do I show all commits in the log view?

### DISCUSSION THREAD

**Discussion by star-szr** [NeogitOrg/neogit#814] [Q&A] [answered]
I know there is `=n Limit number of commits (--max-count=256)` in the log popup, but how do I disable this altogether? 

Yes, I can change the number to something really high and hope for the best, but sometimes I really don't want this flag at all. Is this possible? Programmatically is fine since I'm planning to make this into a mapping anyway.

**Comment by CKolkey** [comment, marked as answer]
Hey @star-szr,

I made the choice to prevent that option from being disabled because the performance of the log buffer isn't _great_ with thousands of commits. So, to prevent users from having a Bad Time™️, I figured that it wouldn't be a big issue to just add a default.

Out of curiosity, if you set the value to something like `10,000`, is the buffer performant enough to navigate in?

**Reply by star-szr** [reply]
Thanks! I see what you mean, even with 1000 items expanding/collapsing the details of each commit (with `<tab>`) is noticeably slower compared to a smaller set of commits.

And of course just generating the log can take more than a few seconds, running into #813.

I think this is OK, I don't have many use cases for viewing a massive amount of commits and I should probably just get in the habit of using the search options more (`--grep` specifically).
